### ISSUE TITLE
Accept Dexcom Share Trend as String

### ISSUE THREAD

**Issue by marionbarker** [loopnlearn/LoopWorkspace#18]
**Describe the bug**
Dexcom modified their Share API to provide Trend as a string rather than an int.

A hotfix branch for FreeAPS has been created.  Once this is tested, this will be merged into the current release of FreeAPS.

Volunteers - please test this:

Step 1: Using released FreeAPS (v2.2 (204) or earlier)
  * Delete your current CGM and add in Dexcom Share - observe the error (screenshot below)

Step 2: Build the hotfix branch over your app
  * Confirm that Dexcom Share works again and provides appropriate trend arrows

Step 3: Restore your system to desired CGM status - with exception of the hotfix, this branch is identical to the current release

To download the hotfix branch, please copy and paste this in your terminal and build using the standard workspace build method.
```
git clone --branch=hotfix/dexcom-share --recurse-submodules https://github.com/loopnlearn/LoopWorkspace
```

Image before the fix:

![DexcomShare-freeaps](https://user-images.githubusercontent.com/19607791/145488206-e2fd06ad-f372-45ae-903d-c5d5aac15e18.PNG)

Image after the fix:
![DexcomShare-hotfix-dexcom-share](https://user-images.githubusercontent.com/19607791/145488246-27073a10-4e10-4020-9924-7f2830781007.PNG)

## More details:
First it was outside US and now inside US. Because all servers now use the same protocol, just switch to using a string.

Copied the information about string and int conversion from share2nightscout-bridge, who set up capability to handle either string or int in the json.

Copy of the Direction Array from share2nightscout-bridge.
```
var DIRECTIONS = {
  NONE: 0
, DoubleUp: 1
, SingleUp: 2
, FortyFiveUp: 3
, Flat: 4
, FortyFiveDown: 5
, SingleDown: 6
, DoubleDown: 7
, 'NOT COMPUTABLE': 8
, 'RATE OUT OF RANGE': 9
};
```

**Comment by codelinx** [comment]
tested and i now have the functionality returned.

![image](https://user-images.githubusercontent.com/5210394/145535061-a87e1953-9744-4944-ab8d-95f485634599.png)

**Comment by carolvchn** [comment]
Tested. Using US-based share account and iOS 14.8.  Let me know if you need more details.

**Comment by marionbarker** [comment]
I updated the hotfix/dexcom-share branch to be both more elegant (thanks @hpc4fun, in https://github.com/LoopKit/Loop/issues/1576) and more robust (in case Dexcom reverts this change).

**Comment by marionbarker** [comment]
This issue is closed with release/tag v2.2.204.hotfix
https://github.com/loopnlearn/LoopWorkspace/releases/tag/v2.2.204.hotfix
